oak/server/rust/oak_runtime/src/runtime/mod.rs

```diff
@@ -569,8 +578,32 @@ impl Runtime {
 
     /// Remove a [`Node`] by [`NodeId`] from the [`Runtime`].
     pub fn remove_node_id(&self, node_id: NodeId) {
+        {
+            // Close any remaining handles
```

Comment:
@daviddrysdale the ABI tests pass if I disable closing any remaining handles when removing nodes. I think this means the cpp runtime doesn't close any remaining handles on exit (that aren't explicitly closed), I could be wrong. We probably want to close any remaining handles on node exit- WDYT?

Reply:
Yeah, handles should be dropped on node exit, and that's probably missing from the C++ runtime too.  

I don't think there's any way to test for proper cleanup, as yet.  Potentially, if we exposed a ([debug-only](https://github.com/project-oak/oak/issues/672)) [metric](https://github.com/project-oak/oak/issues/736) for (say) node count and channel count we could run a test that checked they exactly matched what was expected, but it would be a pretty brittle test.

Reply:
Since the ABI tests don't pass with this change I guess it causes some channels to become orphaned that previously weren't. I'll investigate and add any test changes to this PR

Reply:
That's odd &ndash; all the ABI tests still passed with the C++ runtime when I dropped channels on Node exit (in #783)

Reply:
Ahh it was just the logging thread closing before the node threads, due to a temporary patch i had in place

Reply:
Thanks